sycl/test/on-device/basic_tests/specialization_constants/non_native/Inputs/common.cpp

```diff
@@ -0,0 +1,54 @@
+#include <sycl/sycl.hpp>
```

Comment:
Do we need a new test? I mean, I like tests, but I would expect https://github.com/intel/llvm/blob/sycl/sycl/test/on-device/basic_tests/specialization_constants/host_apis.cpp to work for all cases, both native and emulated.

Reply:
https://github.com/intel/llvm/blob/sycl/sycl/test/on-device/basic_tests/specialization_constants/host_apis.cpp doesn't work for non-native spec constants due to this limitation in spec:
```
class kernel_bunble {
...
/* Available only when: (State == bundle_state::input) */
template<auto& S>
void set_specialization_constant(
  typename std::remove_reference_t<decltype(S)>::type value);
...
```
AOT binaries have `bundle_state::executable`. It means that for non-native spec constants we can't use `set_specialization_constant` method from `kernel_bundle` class, only `set_specialization_constant` from `handler` class is supported.